### PR TITLE
Fix/CX-6045 increase docker pulls limit

### PR DESCRIPTION
**Pull request by ethan605**
# Problem

Builds on TravisCI can fail with the following error:

```
toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
```

# Solution

- Trigger `npm run mock-server:build` for `NODE_ENV=test` and `TRAVIS_PULL_REQUEST=true` only
- Use Docker registry mirrors

## Checklist

_put `n/a` if item is not relevant to PR changes_

- [n/a] Has the CHANGELOG been updated?
- [n/a] Has the README been updated?
- [n/a] Has the CONTRIBUTING doc been updated?
- [n/a] Has the RELEASE_GUIDELINES been updated?
- [n/a] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [n/a] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [n/a] Have any new automated tests been implemented or the existing ones changed?
- [n/a] Have any new manual tests been written down or the existing ones changed?
- [n/a] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1302-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1302-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1302-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1302-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1302-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1302-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1302-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1302-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1302-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1302-pr-onfido-sdk-ui-onfido.surge.sh

File: .travis.yml [hunk 1]
```diff
@@ -14,12 +14,18 @@ cache:
   directories:
     - 'node_modules'
 before_install:
+  - tmpdaemon=$(mktemp)
+  - sudo jq '."registry-mirrors" += ["https://mirror.gcr.io"]' /etc/docker/daemon.json > $tmpdaemon
+  - sudo mv $tmpdaemon /etc/docker/daemon.json
+  - sudo systemctl daemon-reload
+  - sudo service docker restart
```
Comment (ethan605) [inline .travis.yml:21]: Now Docker on TravisCI is using `mirror.gcr.io`:

swappy-20210112-145038

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1302-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1302-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1302-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1302-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1302-pr-onfido-sdk-ui-onfido.surge.sh

File: .travis.yml (hunk 1 above)
Comment (joninvski) [reply .travis.yml:21]: Cool. Nice solution :+1:

File: .travis.yml [hunk 2]
```diff
@@ -14,12 +14,18 @@ cache:
   directories:
     - 'node_modules'
 before_install:
+  - tmpdaemon=$(mktemp)
+  - sudo jq '."registry-mirrors" += ["https://mirror.gcr.io"]' /etc/docker/daemon.json > $tmpdaemon
```
Comment (joninvski) [inline .travis.yml:18]: you probably do not need `sudo` on this one. But no harm will come

**Review by joninvski — APPROVED** [review]

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1302-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1302-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1302-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by joninvski** [comment]
Merging it, and then checking if the `toomanyrequests` error disapears

**Pull request merged**